engine/engine_test.go

```diff
@@ -112,6 +113,71 @@ func TestQueryExplain(t *testing.T) {
 	}
 }
 
+func TestQueryAnalyze(t *testing.T) {
+	opts := promql.EngineOpts{Timeout: 1 * time.Hour}
+	series := storage.MockSeries(
+		[]int64{240, 270, 300, 600, 630, 660},
+		[]float64{1, 2, 3, 4, 5, 6},
+		[]string{labels.MetricName, "foo"},
+	)
+
+	start := time.Unix(0, 0)
+	end := time.Unix(1000, 0)
+
+	assertCPUTimeNonZero := func(t *testing.T, got *engine.AnalyzeOutputNode) {
+		if got != nil {
+			if got.OperatorTelemetry.CPUTimeTaken() <= 0 {
+				t.Errorf("expected non-zero CPUTime, got %v ", got.OperatorTelemetry.CPUTimeTaken())
+			}
+		}
+	}
+
+	for _, tc := range []struct {
+		query string
```

Comment:
Let's try with some complex queries.

Reply:
:+1: let's add this test in the future to the fuzzers